scripts/gmsa/ci-gmsa.sh

```diff
@@ -0,0 +1,211 @@
+main() {
+    if [[ "$(az group exists --name "${CI_RG}")" == "false" ]]; then
+        echo "Requires pre-requsist that resource group ${CI_RG} exists"
+        exit 1
+    fi
+
+    keyvaultid=$(az keyvault show --name "${GMSA_KEYVAULT}" -g "$CI_RG" --query "id" || true)
+    if [[ -z $keyvaultid ]]; then 
+        echo "Requires pre-requsist that keyvault ${GMSA_KEYVAULT} exists"
+        exit 1
+    fi
+
+    # Give permissions to write to keyvault during the domain creation to create secrets that will be used during test
+    domainPricipalId=$(az identity show --name domain-vm-identity --resource-group "$CI_RG" --query 'principalId' -o tsv || true)
+    domainId=$(az identity show --name domain-vm-identity --resource-group "$CI_RG" --query 'id' -o tsv || true)
+    if [[ -z $domainPricipalId ]]; then
+        echo "Requires pre-requsist that user identity 'domain-vm-identity' exists"
+        exit 1
+    fi
+
+    # the powershell commandlet Get-AzUserAssignedIdentity requires ability to read subid which is granted via this custom role
+    # see the setup-gmsa.sh for custom role creation
+    customSubRole=$(az role assignment list --assignee "$domainPricipalId" --query "[?roleDefinitionName=='gMSA']" || true)
+    if [[ $customSubRole == "[]" ]]; then
+        echo "The domain-vm-identity must have custom role 'gMSA'"
+        exit 1
+    fi
+
+    # this identity needs to be assigned to the the Worker nodes that is labeled during e2e set up.
+    userId=$(az identity show --name gmsa-user-identity --resource-group "$CI_RG" --query 'principalId' -o tsv || true)
+    if [[ -z $userId ]]; then 
+        echo "Requires pre-requsist that user identity 'gmsa-user-identity' exists"
+        exit 1
+    fi
+
+    echo "Pre-reqs are met for creating Domain vm"
+    # the custom-data contains scripts to
+    #  - turn this vm into a domain
+    #  - vm is created in vnet that doesn't overlap with default capz cluster vnets
+    #  - creates a domain admin and gmsa users 
+    #  - uploads secrets to the keyvault
+    #  - creates a gmsa yaml spec in location c:\gmsa\gmsa-cred-spec-gmsa-e2e.yml 
+    # this is a random temp password which gets replaced by the cloudbase-init
+    if [[ "$(az group exists --name "${GMSA_NODE_RG}")" == "false" ]]; then
+        az group create --name "$GMSA_NODE_RG" --location "$AZURE_LOCATION" --tags creationTimestamp="$TIMESTAMP"
+    fi
+
+    winpass=$(openssl rand -base64 32)
+    vmname="dc-${GMSA_ID}"
+    vmid=$(az vm show -n "$vmname" -g "$GMSA_NODE_RG" --query "id" || true)
+    if [[ -z $vmid ]]; then 
+        echo "Creating Domain vm"
+        GMSA_DOMAIN_ENVSUBST="${REPO_ROOT}/scripts/gmsa/domain.init"
+        GMSA_DOMAIN_FILE="${REPO_ROOT}/scripts/gmsa/domain.init.tmpl"
+        $ENVSUBST < "$GMSA_DOMAIN_FILE" > "$GMSA_DOMAIN_ENVSUBST"
+        az vm create -l "$AZURE_LOCATION" -g "$GMSA_NODE_RG" -n "$vmname" \
```

Comment:
did you consider using the go sdk to run these prerequisites directly in the test suite instead of using the az cli in a script? similar to what we do for private cluster custom vnet setup